tinygrad/codegen/linearizer.py

```diff
@@ -19,6 +19,11 @@ class UOps(Enum):
   LOAD = auto(); STORE = auto(); CONST = auto(); BARRIER = auto(); PHI = auto() # noqa: E702
   ALU = auto(); WMMA = auto(); CAST = auto(); GEP = auto() # noqa: E702
 
+# TODO temp util
+def dt(dtype: DType, amt=1):
+  if isinstance(dtype, ImageDType): dtype = dtype.base
```

Comment:
is this what `dtype.scalar()` tries to unify?

Reply:
i dont think it's related to that

imagedtype being the same as float should only be a linearizer thing

scalar is a generic vectorized<->normal dtypes bridge

Reply:
i meant write this line as `dtype = dtype.scalar()`, instead of `if isinstance...`